api/clients/v2/examples/client_construction.go

```diff
@@ -142,8 +185,13 @@ func createValidatorPayloadRetriever() (*payloadretrieval.ValidatorPayloadRetrie
 		return nil, fmt.Errorf("create eth client: %w", err)
 	}
 
+	contractDirectory, err := createContractDirectory(context.Background(), logger, ethClient)
+	if err != nil {
+		return nil, fmt.Errorf("create contract directory: %w", err)
+	}
+
```

Comment:
nit: not related to this PR but I've noticed in general we seem to call the contractDirectory at random places whenever an address is needed.

I think overall I would prefer if we fetched all addresses that we need at the very beginning of build processes (whether this be in prod code, or test code, or examples like this) for consistency. This way we know from the start if it fails and theres a config issue. Also I prefer the dependency injection way of passing down the address that a service needs only, not the directory and then letting each contract understand how to fetch the addresses it needs. Thoughts?